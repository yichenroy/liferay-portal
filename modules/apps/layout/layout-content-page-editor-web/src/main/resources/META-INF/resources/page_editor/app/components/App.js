/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import PropTypes from 'prop-types';
import React, {useEffect} from 'react';

import useAutoExtendSession from '../../core/hooks/useAutoExtendSession';
import {INIT} from '../actions/types';
import {config} from '../config/index';
import {reducer} from '../reducers/index';
import {StoreContextProvider, useSelector} from '../store/index';
import {DragAndDropContextProvider} from '../utils/dragAndDrop/useDragAndDrop';
import {CollectionActiveItemContextProvider} from './CollectionActiveItemContext';
import {ControlsProvider} from './Controls';
import DragPreview from './DragPreview';
import LayoutViewport from './LayoutViewport';
import Sidebar from './Sidebar';
import Toolbar from './Toolbar';
import URLParser from './URLParser';

export default function App({state}) {
	const initialState = reducer(state, {type: INIT});

	useAutoExtendSession();

	return (
		<StoreContextProvider initialState={initialState} reducer={reducer}>
			<LanguageDirection />
			<URLParser />
			<ControlsProvider>
				<CollectionActiveItemContextProvider>
					<DragAndDropContextProvider>
						<DragPreview />
						<Toolbar />
						<LayoutViewport />
						<Sidebar />
					</DragAndDropContextProvider>
				</CollectionActiveItemContextProvider>
			</ControlsProvider>
		</StoreContextProvider>
	);
}

App.propTypes = {
	state: PropTypes.object.isRequired,
};

const LanguageDirection = () => {
	const languageId = useSelector((state) => state.languageId);

	useEffect(() => {
		const currentLanguageDirection = config.languageDirection[languageId];
		const wrapper = document.getElementById('wrapper');

		if (wrapper) {
			wrapper.dir = currentLanguageDirection;
			wrapper.lang = languageId;
		}
	}, [languageId]);

	return null;
};
