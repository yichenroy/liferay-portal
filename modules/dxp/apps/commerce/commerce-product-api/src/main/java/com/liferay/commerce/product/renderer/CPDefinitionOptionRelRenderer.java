/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}