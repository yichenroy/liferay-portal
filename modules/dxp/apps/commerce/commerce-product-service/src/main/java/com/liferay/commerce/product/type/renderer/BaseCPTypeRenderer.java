/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}