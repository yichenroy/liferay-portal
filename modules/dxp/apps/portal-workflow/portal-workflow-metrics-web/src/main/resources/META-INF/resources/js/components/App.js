import fetch from '../shared/rest/fetch';
import ProcessListCard from './process-list/ProcessListCard';
import React from 'react';
import Router from '../shared/components/router/Router';
import SLAForm from './sla/SLAForm';
import SLAListCard from './sla/SLAListCard';

/**
 * @class
 * @classdesc Application starter.
 */
export default class AppComponent extends React.Component {
	render() {
		const {companyId, defaultPath = 'process-list'} = this.props;

		return (
			<div className="portal-workflow-metrics-app">
				<Router
					defaultPath={defaultPath}
					paths={[
						{
							component: props => (
								<SLAForm {...props} client={fetch} companyId={companyId} />
							),
							path: 'sla-form',
							title: Liferay.Language.get('new-sla')
						},
						{
							component: props => (
								<SLAListCard {...props} client={fetch} companyId={companyId} />
							),
							path: 'sla-list',
							title: Liferay.Language.get('slas')
						},
						{
							component: props => (
								<ProcessListCard
									{...props}
									client={fetch}
									companyId={companyId}
								/>
							),
							path: 'process-list',
							title: Liferay.Language.get('metrics')
						}
					]}
				/>
			</div>
		);
	}
}