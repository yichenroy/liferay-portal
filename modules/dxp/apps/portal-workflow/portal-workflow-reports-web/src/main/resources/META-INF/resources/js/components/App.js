import graphqlClient from '../shared/apollo/client';
import ProcessListCard from './process-list/ProcessListCard';
import React from 'react';

export default class AppComponent extends React.Component {
	render() {
		const {companyId} = this.props;
		return (
			<div className="portal-workflow-reports-app">
				<ProcessListCard client={graphqlClient} companyId={companyId} />
			</div>
		);
	}
}