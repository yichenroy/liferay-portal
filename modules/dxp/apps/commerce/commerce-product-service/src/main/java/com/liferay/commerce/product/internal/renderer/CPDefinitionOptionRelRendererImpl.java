/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}