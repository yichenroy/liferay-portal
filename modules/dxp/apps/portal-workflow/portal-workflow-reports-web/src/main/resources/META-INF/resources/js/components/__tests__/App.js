import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
	const component = renderer.create(<App />);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});