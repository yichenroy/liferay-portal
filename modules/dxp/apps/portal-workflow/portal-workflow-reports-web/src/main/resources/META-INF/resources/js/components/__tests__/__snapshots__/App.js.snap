// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="portal-workflow-reports-app container-fluid"
>
  <div
    className="col-12"
  >
    <div
      className="row card-panel"
    >
      <div
        className="col-9"
      >
        <div
          className="card"
        >
          <div
            className="card-header bg-transparent border-secondary card-header-default text-secondary semi-bold"
          >
            Open Processes Summary
          </div>
          <div
            className="card-body"
          >
            <div
              className="row d-flex justify-content-start"
              style={
                Object {
                  "marginTop": "8px",
                }
              }
            >
              <div
                className="summary-card border col-2"
                style={
                  Object {
                    "marginLeft": "16px",
                  }
                }
              >
                <span
                  className="regular-text semi-bold text-secondary"
                >
                  Total Open
                </span>
                <div
                  className=""
                >
                  <span
                    className="font-weight-normal"
                    style={
                      Object {
                        "fontSize": "2.5rem",
                      }
                    }
                  >
                    15
                  </span>
                  <span
                    className="regular-text text-secondary"
                  >
                     
                    items
                  </span>
                </div>
              </div>
              <div
                className="summary-card border col-2"
                style={
                  Object {
                    "marginLeft": "16px",
                  }
                }
              >
                <span
                  className="regular-text semi-bold text-secondary"
                >
                  On time
                </span>
                <div
                  className=""
                >
                  <span
                    className="font-weight-normal"
                    style={
                      Object {
                        "fontSize": "2.5rem",
                      }
                    }
                  >
                    82
                  </span>
                  <span
                    className="regular-text text-secondary"
                  >
                     
                    items
                  </span>
                </div>
              </div>
              <div
                className="summary-card border col-2"
                style={
                  Object {
                    "marginLeft": "16px",
                  }
                }
              >
                <span
                  className="regular-text semi-bold text-secondary"
                >
                  Overdue
                </span>
                <div
                  className=""
                >
                  <span
                    className="font-weight-normal"
                    style={
                      Object {
                        "fontSize": "2.5rem",
                      }
                    }
                  >
                    33
                  </span>
                  <span
                    className="regular-text text-secondary"
                  >
                     
                    items
                  </span>
                </div>
              </div>
            </div>
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                }
              }
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <div>
          <nav
            className="management-bar management-bar-light navbar navbar-expand-md"
          >
            <div
              className="container"
            >
              <ul
                className="navbar-nav"
              >
                <li
                  className="dropdown nav-item"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    className="dropdown-toggle nav-link navbar-breakpoint-down-d-none"
                    data-toggle="dropdown"
                    href="#1"
                    role="button"
                  >
                    <span
                      className="navbar-text-truncate"
                    >
                      Filter and Order2
                    </span>
                    <svg
                      className="lexicon-icon lexicon-icon-caret-bottom"
                      focusable="false"
                      role="presentation"
                    >
                      <use
                        href="/images/icons/icons.svg#caret-bottom"
                      />
                    </svg>
                  </a>
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    className="nav-link nav-link-monospaced dropdown-toggle navbar-breakpoint-d-none"
                    data-toggle="dropdown"
                    href="#1"
                    role="button"
                  >
                    <svg
                      className="lexicon-icon lexicon-icon-filter"
                      focusable="false"
                      role="presentation"
                    >
                      <use
                        href="/images/icons/icons.svg#filter"
                      />
                    </svg>
                  </a>
                  <div
                    className="dropdown-menu"
                    role="menu"
                  >
                    <ul
                      className="list-unstyled"
                    >
                      <li>
                        <a
                          className="dropdown-item"
                          href="#1"
                          role="button"
                        >
                          Filter Action 1
                        </a>
                      </li>
                      <li>
                        <a
                          className="dropdown-item"
                          href="#1"
                          role="button"
                        >
                          Filter Action 2
                        </a>
                      </li>
                      <li>
                        <a
                          className="dropdown-item"
                          href="#1"
                          role="button"
                        >
                          Filter Action 3
                        </a>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
              <div
                className="navbar-form navbar-form-autofit navbar-overlay navbar-overlay-sm-down"
              >
                <div
                  className="container"
                >
                  <form
                    role="search"
                  >
                    <div
                      className="input-group"
                    >
                      <div
                        className="input-group-item"
                      >
                        <input
                          className="form-control input-group-inset input-group-inset-after"
                          placeholder="Search for..."
                          type="text"
                        />
                        <span
                          className="input-group-inset-item input-group-inset-item-after"
                        >
                          <button
                            className="btn btn-unstyled navbar-breakpoint-d-none"
                            type="button"
                          >
                            <svg
                              className="lexicon-icon lexicon-icon-times"
                              focusable="false"
                              role="presentation"
                            >
                              <use
                                href="/images/icons/icons.svg#times"
                              />
                            </svg>
                          </button>
                          <button
                            className="btn btn-unstyled navbar-breakpoint-d-block"
                            type="button"
                          >
                            <svg
                              className="lexicon-icon lexicon-icon-search"
                              focusable="false"
                              role="presentation"
                            >
                              <use
                                href="/images/icons/icons.svg#search"
                              />
                            </svg>
                          </button>
                        </span>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <div
          className="table-responsive"
        >
          <table
            className="show-quick-actions-on-hover table table-autofit table-hover table-list table-nowrap"
          >
            <thead>
              <tr>
                <th
                  className="table-cell-expand table-head-title"
                  style={
                    Object {
                      "width": "70%",
                    }
                  }
                >
                  Process Name
                </th>
                <th
                  className="table-cell-expand table-head-title"
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  On Time
                </th>
                <th
                  className="text-body table-cell-expand table-head-title"
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Overdue
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  Sales Quoting
                </td>
                <td>
                  82 items
                </td>
                <td>
                  33 items
                </td>
              </tr>
              <tr>
                <td>
                  Refund Request
                </td>
                <td>
                  12 items
                </td>
                <td>
                  25 items
                </td>
              </tr>
              <tr>
                <td>
                  Travel Reimbursement
                </td>
                <td>
                  5 items
                </td>
                <td>
                  23 items
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
